BEST_PRACTICE.md

```diff
@@ -0,0 +1,247 @@
+# Clib Examples and Best Practice
+
+This page will cover:
+
+ - [How to use libraries](#how-to-use-installed-libraries).
+ - [Example Makefile](#example-makefile).
+ - [Example `package.json` for executables](#example-packagejson-for-exacutables).
+ - [Making your own library package](#making-your-own-libraries).
+ - [Example `package.json` for libraries](#example-packagejson-for-libraries).
+ - [How to install/uninstall executables](#install-and-uninstall-executables).
+
+For instructions on installation, click [here](https://github.com/clibs/clib#installation).
+
+## How to use installed libraries for your project
+
+Lets say you have your project, with a typical directory tree like this:
+
+```
+your-project/
+│
+├── deps/
+│   ├── trim.c/
+│   │   ├── trim.h
+│   │   ├── trim.c
+│   │   └── package.json
+│   │
+│   ├── commander/
+│   │   ├─ commander.h
+│   │   ├─ commander.c
+│   │   └─ package.json
+│   │
+│   └── logger/
+│       ├── logger.h
+│       ├── logger.c
+│       └── package.json
+│
+├── LICENSE
+│
+├── Makefile
+│
+├── README.md
+│
+├── package.json
+│
+├── src/
+│   ├── main.c
+│   ├── function.c
+│   ├── function.h
+│   └── etc...
+│
+└── test.sh
+```
+
+`test.sh` is your test script. `src` is the directory where you source
+code is in. And `deps` is the directory where you libraries will be
+downloaded when you run `clib install <username/library>`.
+
+Knowing all of that, lets have a look at an example Makefile.
+
+### Example Makefile
+
+```makefile
+# your c compiler
+CC = gcc
+
+# where to install
+PREFIX = /usr/local/bin
+
+# your output file
+TARGET = project-name
+
+# your project name
+TARGET = your-project
+
+CFLAGS = -Ideps -Wall
+
+# all the source files
+SRC = $(wildcard src/*.c)
+SRC += $(wildcard deps/*/*.c)
+
+OBJS = $(SRC:.c=.o)
+
+.PHONY:
+all: $(TARGET)
+
+.PHONY:
+$(TARGET): $(OBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $(TARGET) $(OBJS)
+
+.PHONY:
+%.o: %.c
+	$(CC) $(DEP_FLAG) $(CFLAGS) $(LDFLAGS) -o $@ -c $<
+
+.PHONY:
+clean:
+	rm -f $(OBJS)
+
+.PHONY:
+install: $(TARGET)
+	cp -f $(TARGET) $(PREFIX)
+
+.PHONY:
+uninstall: $(PREFIX)/$(TARGET)
+	rm -f $(PREFIX)/$(TARGET)
+```
+
+This is a basic Makefile, and should work for most of your projects.
```

Comment:
Might be nice to just link to an existing Makefile for this, or use a Gist.

Reply:
I'll be fine with that, although, if I was a user, I prefer to see the Makefile right here. But if you have a Gist Makefile example then let me know. _it should be under your user, rather then mine :wink:_